pages/api/profile.ts

```diff
@@ -0,0 +1,54 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import cache from '../../services/cache';
+import proton from '../../services/proton-rpc';
+
+interface MyAssetRequest extends NextApiRequest {
+  query: {
+    accounts: string[];
+  };
+}
+
+const handler = async (
+  req: MyAssetRequest,
+  res: NextApiResponse
+): Promise<void> => {
+  const {
+    method,
+    query: { accounts },
+  } = req;
+  switch (method) {
+    case 'POST':
+      break;
+    case 'PUT':
+      break;
+    case 'PATCH':
+      break;
+    default: {
+      try {
+        const chainAccounts =
+          typeof accounts === 'string' ? [accounts] : [...new Set(accounts)];
+
+        for (const account of chainAccounts) {
+          if (!cache.has(account)) {
+            const res = await proton.getProfileImage({ account });
```

Comment:
When I console.log in here, and I refresh the page, leave the page, and come back to the page, it still generates the same amount of console logs. See logs here:

![Screen Shot 2021-03-23 at 4 19 03 PM](https://user-images.githubusercontent.com/18355075/112231249-98b14400-8bf3-11eb-8033-5085c079af64.png)

If the cache is truly working, then we should not see any duplicate names on this list.

Reply:
Issue has been resolved! See PR description for testing steps.